quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/FeaturesDistTest.java

```diff
@@ -52,10 +53,18 @@ public void testEnablePreviewFeatures(LaunchResult result) {
 
     @Test
     @Launch({StartDev.NAME, "--features=preview", "--features-disabled=token-exchange"})
+    public void testPreviewFeatureDisabledInPreviewMode(LaunchResult result) {
+        CLIResult cliResult = (CLIResult) result;
+        cliResult.assertStartedDevMode();
+        assertFalse(cliResult.getOutput().contains("token-exchange"));
+    }
+
+    @Test
+    @Launch({StartDev.NAME, "--features=token-exchange", "--features-disabled=token-exchange"})
     public void testEnablePrecedenceOverDisable(LaunchResult result) {
         CLIResult cliResult = (CLIResult) result;
         cliResult.assertStartedDevMode();
-        assertPreviewFeaturesEnabled((CLIResult) result);
+        assertThat(cliResult.getOutput(), containsString("Preview feature enabled: token_exchange"));
     }
```

Comment:
This used to test that token-exchange was enabled with `--features=preview --features-disabled=token-exchange`. That's not ideal, as it should be possible to disable one or two features from the preview profile.

Added a separate test to check if a feature is both enabled and disabled it will be enabled.